### PR TITLE
File-versioning feature

### PR DESCRIPTION
**Pull request by felipemorais**
Add unique parameter like in #26

### CODE REVIEW

File: lib/basket.js [hunk 1]
```diff
@@ -107,7 +107,11 @@
 		obj.execute = obj.execute !== false;
 		scripts[ scriptIndex ] = null;
 		
-		if( !source || source.expire - +new Date() < 0 ) {
+		if( !source || source.expire - +new Date() < 0  || obj.unique !== source.unique) {
```
Comment (sindresorhus) [inline lib/basket.js]: `if (`

File: lib/basket.js [hunk 2]
```diff
@@ -107,7 +107,11 @@
...
+			if( obj.unique ) {
+				//set parameter to prevent browser cache
```
Comment (sindresorhus) [inline lib/basket.js]: `// s`

File: lib/basket.js [hunk 3]
```diff
@@ -107,7 +107,11 @@
...
+				obj.url += ( (obj.url.indexOf("?") > 0 ) ? "&" : "?" ) + "basket-unique="+obj.unique;
```
Comment (sindresorhus) [inline lib/basket.js]: single quotes
space between operators

File: test/tests.js [hunk 4]
```diff
@@ -132,3 +132,70 @@ asyncTest( 'store data using expiration (expired)', 2, function() {
...
+			.wait(function() {
+				var stamp = basket.get('fixtures/stamp-script.js').stamp;
+				ok( basket.get('fixtures/stamp-script.js'), 'Data exists in localStorage' );
+
+				basket
+					.require({ url: 'fixtures/stamp-script.js', unique: 456 })
+					.wait(function() {
+						var req = basket.get('fixtures/stamp-script.js');
+						ok( stamp !== req.stamp, 'Data retrieved from server' );
+						ok( req.url.indexOf("basket-unique=456") > 0, 'Sending basket unique parameter' );
```
Comment (sindresorhus) [inline test/tests.js]: single quotes

File: test/tests.js [hunk 5]
```diff
@@ -132,3 +132,70 @@ asyncTest( 'store data using expiration (expired)', 2, function() {
...
+			.wait(function() {
+				var stamp = basket.get('fixtures/stamp-script.js').stamp;
+				ok( basket.get('fixtures/stamp-script.js'), 'Data exists in localStorage' );
+
+				basket
+					.require({ url: 'fixtures/stamp-script.js', unique: 123 })
+					.wait(function() {
+						var req = basket.get('fixtures/stamp-script.js');
+						ok( stamp !== req.stamp, 'Data retrieved from server' );
+						ok( req.url.indexOf("basket-unique=123") > 0, 'Sending basket unique parameter' );
```
Comment (sindresorhus) [inline test/tests.js]: single quotes

**Comment by sindresorhus** [comment]
Does the testsuite pass locally?

Can we have some docs?

**Comment by felipemorais** [comment]
Yes, pass!

Docs is very outdated.
I'm updating in another branch.
Problem?

**Comment by sindresorhus** [comment]
Not at all

**Comment by sindresorhus** [comment]
Landed. Thanks :)

**Pull request closed without merging**